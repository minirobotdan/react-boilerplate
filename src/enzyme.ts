
import * as Enzyme from 'enzyme'
import { configure, shallow, mount, render } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;