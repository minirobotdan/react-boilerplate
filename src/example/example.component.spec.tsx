import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from '../enzyme';
import { ExampleComponent } from './example.component';


test('Renders correctly', () => {
    const component = renderer.create(
        <ExampleComponent foo="world"></ExampleComponent>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Displays the correct message, based on passed in props', () => {
    const wrapper = shallow(<ExampleComponent foo="world"></ExampleComponent>);
    expect(wrapper.contains(<h1>Hello, world!</h1>));
});

test('The counter increases when the button is clicked', () => {
    const component = shallow(<ExampleComponent foo="world"></ExampleComponent>);
    expect(component.contains(<button >Count: 0</button>));
    component.find('button').at(0).simulate('click');
    expect(component.contains(<button >Count: 1</button>));
});
