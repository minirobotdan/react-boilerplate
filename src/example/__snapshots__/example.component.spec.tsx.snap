// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="sc-bdVaJa eypWwp"
>
  <h1>
    Hello, world!
  </h1>
  <button
    onClick={[Function]}
  >
    Count: 
    0
  </button>
</div>
`;
